crates/shared/src/price_estimation/factory.rs

```diff
@@ -268,15 +267,22 @@ impl<'a> PriceEstimatorFactory<'a> {
                 rate_limiting_strategy,
                 format!("{}_estimator", driver.name),
             ));
-            let estimator = Arc::new(ExternalTradeFinder::new(
+            let estimator = Arc::new(ExternalPriceEstimator::new(
                 driver.url.clone(),
                 self.components.http_factory.create(),
                 rate_limiter,
+                self.network.settlement,
             ));
+            let verified = self.trade_verifier.as_ref().map(|trade_verifier| {
+                let estimator = estimator.clone().verified(trade_verifier.clone());
+                instrument(estimator, format!("{}_verified", driver.name)) as _
+            });
+
             let entry = EstimatorEntry {
                 optimal: estimator.clone(),
                 fast: estimator.clone(),
                 native: estimator,
+                verified,
             };
```

Comment:
Specifically, this code can be shared with `create_estimator_entry`. (see comment https://github.com/cowprotocol/services/pull/1512/files#r1198810411)